app/models/event.rb

```diff
@@ -11,11 +11,29 @@
 #  updated_at      :datetime         not null
 #  eventable_id    :bigint
 #  organization_id :bigint
+#  user_id         :bigint
 #
 class Event < ApplicationRecord
   scope :for_organization, ->(organization_id) { where(organization_id: organization_id).order(:event_time) }
 
   serialize :data, EventTypes::StructCoder.new(EventTypes::InventoryPayload)
 
   belongs_to :eventable, polymorphic: true
+  belongs_to :user, optional: true
+
+  before_create do
+    self.user_id = PaperTrail.request&.whodunnit
```

Comment:
Ah, as a mechanism to get the current user, I see. Seems fine.

Reply:
I wonder if there are other things to tie to, like a request id that goes into the logs. Other stuff to make tracing of what went wrong easier

Reply:
I was looking into that... I can try to add more metadata to PaperTrail, perhaps. Otherwise you need to mess with thread variables and it gets messy. The truth is that the operation itself should give us most if not everything that we need - there are very few ways (usually one) that any individual event can be created. If we find some pain dealing with the current way, we can always add more info later.